components/dashboard/src/admin/UserDetail.tsx

```diff
@@ -0,0 +1,238 @@
+                                }
+                            }}
+                        >{user.featureFlags?.permanentWSFeatureFlags?.join(', ') || '---'}</Property>
+                        <Property name="Roles"
+                            action={{
+                                label: 'Edit Roles',
+                                onClick: () => {
+                                    setEditRoles(true);
+                                }
+                            }}
+                        >{user.rolesOrPermissions?.join(', ') || '---'}</Property>
+                        <Property name="Student">{isStudent === undefined ? '---' : (isStudent ? 'Enabled' : 'Disabled')}</Property>
+                    </div>
+                </div>
+            </div>
+            <WorkspaceSearch user={user} />
+        </PageWithSubMenu>
+        <Modal visible={editFeatureFlags} onClose={() => setEditFeatureFlags(false)} title="Edit Feature Flags" buttons={[
+            <button className="secondary" onClick={() => setEditFeatureFlags(false)}>Done</button>
+        ]}>
+            <p>Edit feature access by adding or removing feature flags for this user.</p>
+            <div className="flex flex-col">
+                {
+                    flags.map(e => <CheckBox key={e.title} title={e.title} desc="" checked={!!e.checked} onChange={e.onClick} />)
+                }
+            </div>
+        </Modal>
+        <Modal visible={editRoles} onClose={() => setEditRoles(false)} title="Edit Roles" buttons={[
+            <button className="secondary" onClick={() => setEditRoles(false)}>Done</button>
+        ]}>
+            <p>Edit user permissions by adding or removing roles for this user.</p>
+            <div className="flex flex-col">
+                {
+                    rop.map(e => <CheckBox key={e.title} title={e.title} desc="" checked={!!e.checked} onChange={e.onClick} />)
+                }
+            </div>
+        </Modal>
+        <Modal visible={viewAccountStatement} onClose={() => setViewAccountStatement(false)} title="Edit Roles" buttons={[
+            <button className="secondary" onClick={() => setViewAccountStatement(false)}>Done</button>
+        ]}>
+            <div className="flex flex-col">
+                {
+                    JSON.stringify(accountStatement, null, '  ')
+                }
+            </div>
+        </Modal>
+    </>;
+}
+
+function Label(p: { text: string, color: string }) {
+    return <div className={`ml-3 text-sm text-${p.color}-600 truncate bg-${p.color}-100 px-1.5 py-0.5 rounded-md my-auto`}>{p.text}</div>;
+}
+
+export function Property(p: { name: string, children: string | ReactChild, action?: { label: string, onClick: () => void } }) {
+    return <div className="ml-3 flex flex-col w-4/12 truncate">
+        <div className="text-base text-gray-500 truncate">
+            {p.name}
+        </div>
+        <div className="text-lg text-gray-600 font-semibold truncate">
```

Comment:
**issue**: We could truncate this sooner so that it does not break the rest of the interface when the list is long.

<img width="903" alt="Screenshot 2021-04-05 at 11 30 33 AM" src="https://user-images.githubusercontent.com/120486/113554731-701d4700-9602-11eb-889c-7fa7579a0026.png">

Reply:
Hey George 👋 you look so good these days 😁

Reply:
Feels comfortable over here! 😅